mesh/mesh.cpp

```diff
@@ -6653,6 +6656,242 @@ void Mesh::GetEdgeToUniqueKnotvector(Array<int> &edge_to_ukv,
+         bool hasKV = false;
+         for (auto e : fe)
+         {
+            const int skv = edge_to_ukv[e];
+            if (skv == kv || sign(skv) == kv) { hasKV = true; }
+         }
+         if (!hasKV) { continue; }
+
+         Array<int> row;
+         face2elem->GetRow(face, row);
+         for (auto elem : row) { nghb.insert(elem); }
+      }
+   };
+
+   std::vector<std::vector<int>> dir_edges;
+   if (dim == 2)
+   {
+      dir_edges =
+      {
+         {0,2},
+         {1,3}
+      };
+   }
+   else
+   {
+      dir_edges =
+      {
+         {0,2,4,6},
+         {1,3,5,7},
+         {8,9,10,11}
+      };
+   }
+
+   Array<int> ukvs((dim==2) ? 4 : 12);
+   Array<int> pe, oe;
+   bool initKV = false;
+
+   auto setPatchDirections = [&](int p, int kv, Array<bool> &edgeSet,
+                                 std::unordered_set<int> &visited)
+   {
+      // Edges and orientations for this patch
+      GetElementEdges(p, pe, oe);
+
+      // Get the signed unique knot vector indices
+      for (int i = 0; i < pe.Size(); i++)
+      {
+         ukvs[i] = edge_to_ukv[pe[i]];
+         ukvs[i] = (oe[i] < 0) ? sign(ukvs[i]) : ukvs[i];
+      }
+
+      // Find the dimension with this kv.
+      int thisDim = -1;
+      for (int d=0; d<dim; ++d) // Loop over dimensions.
+      {
+         const int skv = edge_to_ukv[pe[dir_edges[d][0]]];
+         if (skv == kv || sign(skv) == kv)
+         {
+            thisDim = d;
+         }
+      }
+      if (thisDim == -1)
+      {
+         return false;
+      }
+
+      // For this dimension, find any edge already set. If no edge is set, we
+      // arbitrarily take the first.
+      int ref_edge0 = dir_edges[thisDim][0];
+      for (auto ref_edge : dir_edges[thisDim])
+      {
+         const int edge = pe[ref_edge];
+         if (edgeSet[edge])
+         {
+            ref_edge0 = ref_edge;
+         }
+      }
+
+      if (initKV && !edgeSet[pe[ref_edge0]])
+      {
+         visited.erase(p);
+         return false; // There is no set edge in this direction on this patch.
+      }
+
+      initKV = true;
+
+      // Use ref_edge0 to set other edges in this dimension.
+      edgeSet[pe[ref_edge0]] = true;
+      for (auto i : dir_edges[thisDim])
+      {
+         if (i == ref_edge0)
+         {
+            continue;
+         }
+
+         const int edge = pe[i];
+         if ((dim == 2 && ukvs[i] != sign(ukvs[ref_edge0])) ||
+             (dim == 3 && ukvs[i] == sign(ukvs[ref_edge0])))
+         {
+            // Flip the sign of this edge
+            MFEM_VERIFY(!edgeSet[edge], "");
+            edge_to_ukv[edge] = sign(edge_to_ukv[edge]);
+         }
+
+         edgeSet[edge] = true;
+      }
+
+      return true;
+   };
+
+   Array<bool> edgeSet(NumOfEdges);
+   edgeSet = false;
+
+   std::unordered_set<int> unset;
+   for (int i=0; i<NumOfElements; ++i) { unset.insert(i); }
+
+   const int max_iter = 3 * NumOfElements;
+   for (int iter=0; iter<max_iter; ++iter)
```

Comment:
I think I roughly understand what this loop will do.
But I think the  this entire routine, and this loop in particular could benefit from extra explanation. How do the seet, unset, visited atc flags work? What are the assumptions. My guess is that Mobius strip like meshes will not work  for this routine as they create an error, infinite loop, or an end result that does not pass the test. This is worth checking..
This is one of the reasons I have not started on multipatch VectorFE.

So perhaps this reasonable limitation  (if it is indeed the case) should be documented.